### PR TITLE
First pass of direct db calls for origins

### PR DESCRIPTION
**Pull request by eeyun**
All functional tests should be passing.

This... was another gnarly one.

Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: test/builder-api/src/origins.js [hunk 1]
```diff
@@ -33,7 +33,12 @@ describe('Origin API', function () {
       request.get('/depot/origins/neurosis')
         .expect(200)
         .end(function (err, res) {
-          expect(res.body).to.deep.equal(global.originNeurosis);
+          console.log(res.body);
```
Comment (elliott-davis) [inline test/builder-api/src/origins.js]: remove console log

File: components/builder-api/src/server/resources/origins.rs [hunk 2]
```diff
@@ -35,28 +36,18 @@ use hab_net::{ErrCode, NetOk};
 
 use protocol::originsrv::*;
 
+use db::models::origin::{
+    CreateOrigin as CreateOriginMod, CreateOriginHandlerReq, GetOrigin as GetOriginMod,
```
Comment (chefsalim) [inline components/builder-api/src/server/resources/origins.rs]: Do we need the aliases?

File: components/builder-api/src/server/resources/origins.rs [hunk 3]
```diff
@@ -173,77 +164,91 @@ impl Origins {
...
-            .header(http::header::CACHE_CONTROL, headers::NO_CACHE)
-            .json(origin),
-        Err(err) => err.into(),
+    match OriginMod::get(
+        GetOriginMod {
+            name: origin_name.clone(),
+        },
+        &*conn,
+    ) {
+        Ok(origin) => HttpResponse::Ok().json(origin),
```
Comment (chefsalim) [inline components/builder-api/src/server/resources/origins.rs]: We don't need the no_cache?

File: components/builder-db/src/models/origin.rs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    pub owner_id: i64,
+    pub name: String,
+    pub created_at: Option<NaiveDateTime>,
+    pub updated_at: Option<NaiveDateTime>,
+    pub default_package_visibility: PackageVisibility,
+}
+
+#[derive(Debug, Serialize, Deserialize, QueryableByName)]
+#[table_name = "origins_with_secret_key_full_name_v2"]
+pub struct OriginsWithPrivateKey {
```
Comment (chefsalim) [inline components/builder-db/src/models/origin.rs]: should this be named "OriginWithPrivateKey"?

File: components/builder-db/src/schema/origin.rs [hunk 5]
```diff
@@ -1,22 +1,37 @@
...
         name -> Text,
         owner_id -> BigInt,
-        default_package_visibility -> Text,
+        default_package_visibility -> PackageVisibilityMapping,
         created_at -> Nullable<Timestamptz>,
         updated_at -> Nullable<Timestamptz>,
         package_count -> BigInt,
     }
 }
+table! {
```
Comment (chefsalim) [inline components/builder-db/src/schema/origin.rs]: white sp

File: components/builder-db/src/models/origin.rs [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+#[derive(Clone, Serialize, Deserialize)]
+pub struct CreateOrigin {
+    pub name: String,
+    pub owner_id: i64,
+    pub owner_name: String,
+    pub default_package_visibility: PackageVisibility,
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct CreateOriginHandlerReq {
```
Comment (chefsalim) [inline components/builder-db/src/models/origin.rs]: I don't think we should have handler structs intermingled with the db models

File: components/builder-api/src/server/resources/origins.rs (hunk 2 above)
Comment (elliott-davis) [reply components/builder-api/src/server/resources/origins.rs]: unfortunately yes, there are conflicting names in the originsrv protos

File: components/builder-api/src/server/resources/origins.rs (hunk 3 above)
Comment (elliott-davis) [reply components/builder-api/src/server/resources/origins.rs]: we do

File: test/builder-api/src/origins.js [hunk 7]
```diff
@@ -33,7 +33,10 @@ describe('Origin API', function () {
       request.get('/depot/origins/neurosis')
         .expect(200)
         .end(function (err, res) {
-          expect(res.body).to.deep.equal(global.originNeurosis);
+	      expect(res.body.name).to.equal(global.originNeurosis.name);
```
Comment (elliott-davis) [inline test/builder-api/src/origins.js]: I don't even want to type this, but...alas - whitespace

Comment (eeyun) [reply test/builder-api/src/origins.js]: oh dang!

Comment (eeyun) [reply test/builder-api/src/origins.js]: Actually... there isn't extra whitespace there in the code i'm currently looking at, could this be a GH rendering thing?

**Comment by chefsalim** [comment]
Looks like there are some e2e test failures..

**Comment by eeyun** [comment]
There weren't when I tested locally before submitting, but i rebased and now there are 73+ failing tests in APIs that this change didn't touch. Jobs, Channels, Keys, Integrations, Packages and Invitations. Looking into it but, these failures might not be from this change.

**Comment by eeyun** [comment]
Ah actually I do see what's happened, looks like the rebase made a monstrous mess.

**Comment by eeyun** [comment]
We discovered the failures here were related to the outage in builder. Going to refresh the build.

**Comment by elliott-davis** [comment]
This is a unit failure that is fixed in a rebase - it's probably safe to merge

**Pull request merged**